### PR TITLE
[PR](PoC): Added PoC for both network protocol UDP and TCP

### PR DESCRIPTION
**Pull request by ctalneau**
This pull request introduces a proof of concept (PoC) for comparing TCP and UDP protocols in the context of real-time multiplayer gaming, specifically for R-Type. It adds a comprehensive test suite with both TCP and UDP echo servers and clients, simulation of packet loss/head-of-line blocking, and detailed documentation and build instructions. The goal is to empirically demonstrate the advantages of UDP over TCP for low-latency game networking and to provide a technical foundation for protocol selection in the project.

The most important changes are:

**Documentation and Rationale:**
* Added a detailed `README.md` explaining the motivation, test procedures, expected behaviors, results, and recommendations for using UDP over TCP in real-time gaming scenarios, including a summary table and implementation strategy for R-Type.

**Test Infrastructure and Build System:**
* Introduced a new `CMakeLists.txt` for building the TCP and UDP test executables (`tcptest` and `udptest`), with proper compiler flags and threading support.

**TCP Test Implementation:**
* Added `TcpTest.cpp` implementing a TCP echo server and client, with support for simulating head-of-line blocking by dropping a packet, measuring per-packet latency, and printing detailed output for analysis.

**UDP Test Implementation:**
* Added `UdpTest.cpp` implementing a UDP echo server and client, with support for simulating packet loss, measuring per-packet latency, and demonstrating UDP's real-time advantages (no head-of-line blocking, lower latency).

### CODE REVIEW

File: PoC/PoC_Network_Protocol/UdpTest.cpp [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                                 packet_start - start_time).count() << "ms" << std::endl;
+                continue;
+            }
+
+            sendto(sock, message.c_str(), message.length(), 0,
+                  (struct sockaddr *)&serv_addr, sizeof(serv_addr));
+
+            struct timeval tv;
+            tv.tv_sec = 0;
+            tv.tv_usec = 100000;
```
[the remark this answers is not in the document]
Comment (ctalneau) [reply PoC/PoC_Network_Protocol/UdpTest.cpp]: This is a PoC

File: PoC/PoC_Network_Protocol/UdpTest.cpp [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        }
+    }
+
+    void sendPackets(int num_packets, bool simulate_loss = false) {
+        auto start_time = std::chrono::high_resolution_clock::now();
+
+        for (int i = 1; i <= num_packets; ++i) {
+            auto packet_start = std::chrono::high_resolution_clock::now();
+
+            std::string message = "Packet " + std::to_string(i) + " - " + std::string(100, 'X');
```
[the remark this answers is not in the document]
Comment (ctalneau) [reply PoC/PoC_Network_Protocol/UdpTest.cpp]: This is a PoC

File: PoC/PoC_Network_Protocol/UdpTest.cpp [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+            auto packet_latency = std::chrono::duration_cast<std::chrono::microseconds>(packet_end - packet_start);
+
+            if (valread > 0) {
+                std::cout << "Packet " << i << " - Latency: " << packet_latency.count() << "μs - "
+                          << std::string(buffer, valread).substr(0, 60) << "..." << std::endl;
+            } else {
+                std::cout << "Packet " << i << " - TIMEOUT/LOSS after " << packet_latency.count() << "μs" << std::endl;
+            }
+
+            std::this_thread::sleep_for(std::chrono::milliseconds(50));
```
[the remark this answers is not in the document]
Comment (ctalneau) [reply PoC/PoC_Network_Protocol/UdpTest.cpp]: This is a PoC

File: PoC/PoC_Network_Protocol/UdpTest.cpp [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+#include <chrono>
+#include <cstring>
+#include <cstdio>
+
+class UdpEchoServer {
+private:
+    int server_fd;
+    struct sockaddr_in address;
+    int addrlen = sizeof(address);
+    bool running = false;
```
[the remark this answers is not in the document]
Comment (ctalneau) [reply PoC/PoC_Network_Protocol/UdpTest.cpp]: This is a PoC

File: PoC/PoC_Network_Protocol/TcpTest.cpp [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+
+    void sendPackets(int num_packets, bool simulate_drop = false) {
+        auto start_time = std::chrono::high_resolution_clock::now();
+
+        for (int i = 1; i <= num_packets; ++i) {
+            auto packet_start = std::chrono::high_resolution_clock::now();
+
+            std::string message = "Packet " + std::to_string(i) + " - " + std::string(100, 'X');
+
+            if (simulate_drop && i == 5) {
```
[the remark this answers is not in the document]
Comment (ctalneau) [reply PoC/PoC_Network_Protocol/TcpTest.cpp]: This is a PoC

**Review by Arkteus — APPROVED** [review]
lgtm 🎖️
